tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_welfords.h

```diff
@@ -44,300 +44,518 @@ struct FloatBits
+ */
 template <std::size_t reciprocal_size>
-void _calculate_welfords_online_(
-    uint32_t current_sample,
-    const uint32_t final_sample,
-    uint32_t skip_n_rows,
-    const std::array<uint32_t, reciprocal_size>& reciprocal_lut,
-    const bool reformat_dst_to_col_on_end,
-    const bool convert_M2_to_var)
+sfpi_inline void _calculate_welfords_tile_(uint32_t start_idx, const std::array<uint32_t, reciprocal_size>& reciprocal_lut)
 {
-    // Pack the mean into the first face of the mean dst reg. Convert M2 to variance and pack into the first face of the var dst reg.
-    if (convert_M2_to_var)
+    // We load 4 rows of a tile (with 32 columns each) at a time and process them.
+    // To finish the entire tile, we need to repeat this process 8 times.
+
+    // Process the first block (4 rows of 32 columns)
+    _calculate_welfords_block_<reciprocal_size, 0, 0>(start_idx, reciprocal_lut);
+
+    // Repeat this 7 more times to process the remaining blocks
+    _calculate_welfords_block_<reciprocal_size, 0, 1>(start_idx + 4, reciprocal_lut);
+    _calculate_welfords_block_<reciprocal_size, 0, 2>(start_idx + 8, reciprocal_lut);
+    _calculate_welfords_block_<reciprocal_size, 0, 3>(start_idx + 12, reciprocal_lut);
+
+    _calculate_welfords_block_<reciprocal_size, 1, 0>(start_idx + 16, reciprocal_lut);
+    _calculate_welfords_block_<reciprocal_size, 1, 1>(start_idx + 20, reciprocal_lut);
+    _calculate_welfords_block_<reciprocal_size, 1, 2>(start_idx + 24, reciprocal_lut);
+    _calculate_welfords_block_<reciprocal_size, 1, 3>(start_idx + 28, reciprocal_lut);
+}
+
+/*
+ * @brief Calculates the Welford's online algorithm for a tile in the dst reg on a subset of rows.
+ *
+ * This function calculates the Welford's online algorithm for a tile in the dst reg.
+ * It assumes that the current mean and m2 values are placed in LREG4 and LREG5, respectively.
+ * It operates on 32 inputs each from 32 columns, present in a 32x32 tile in dst 0. At the end,
+ * the updated mean and m2 values are stored in LREG4 and LREG5, respectively.
+ * @tparam reciprocal_size The size of the reciprocal lookup table.
+ * @param start_idx The index of the first element in the tile; used to index the reciprocal lookup
+ *                  table.
+ * @param start_row The offset of the row to start from. Only rows starting from this offset are
+ *                   processed in the tile. Should be 0 <= start_row <= 31.
+ * @param num_rows The number of rows to process. Should be 0 <= num_rows <= 32. Also,
+ *                 0 <= start_row + num_rows <= 32.
+ * @param reciprocal_lut The lookup table containing the reciprocals of the sample counts.
+ */
+template <std::size_t reciprocal_size>
+sfpi_inline void _calculate_welfords_partial_tile_(
+    uint32_t start_idx, uint32_t start_row, uint32_t num_rows, const std::array<uint32_t, reciprocal_size>& reciprocal_lut)
+{
+    [[unlikely]] if (num_rows == 0)
```

Comment:
```suggestion
    if (num_rows == 0)
```
These are C++20 extensions, likely and unlikely. LLK should keep compatibility with C++17. We are not yet strictly enforcing this, but maybe we should.